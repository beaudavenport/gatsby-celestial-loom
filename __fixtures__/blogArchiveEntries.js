export default {
  allMarkdownRemark:
  {
    totalCount: 2,
    edges: [{
      node: {
        id: '0bb54b5f-3e2a-5fb5-b1d8-884478cddc40',
        frontmatter: {
          title: 'NORWAC - I AM ON MY WAY',
          publishMonthAndYear: 'May 2019',
          publishDate: 'May 23, 2019',
        },
        fields: { slug: '/blog/norwac-i-am-on-my-way/' },
      },
    },
    {
      node: {
        id: '361c122c-1e55-5ee3-9a9f-935f6c6b2e2f',
        frontmatter: {
          title: 'The Pleiades',
          publishMonthAndYear: 'May 2019',
          publishDate: 'May 21, 2019',
        },
        fields: { slug: '/blog/the-pleiades/' },
      },
    }],
  },
};
