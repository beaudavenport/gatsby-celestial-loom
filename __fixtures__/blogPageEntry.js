export default {
  markdownRemark: {
    html: "<p>I'm on my way to NORWAC - an astrological convention in Seattle, WA</p>",
    frontmatter: {
      image: '/assets/cancer-norwac.jpg',
      publishDate: 'May 23, 2019',
      relatedItems: ['Taurus', 'First House'],
      title: 'NORWAC - I AM ON MY WAY',
    },
  },
};
