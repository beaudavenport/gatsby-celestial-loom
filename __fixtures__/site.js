export default {
  site: {
    siteMetadata: {
      title: 'The Celestial Loom',
    },
  },
};
