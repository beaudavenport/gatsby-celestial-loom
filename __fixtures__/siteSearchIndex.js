export default {
  siteSearchIndex: {
    index: null,
  },
};
