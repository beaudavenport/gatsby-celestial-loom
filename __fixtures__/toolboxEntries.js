export default {
  allMarkdownRemark: {
    edges: [
      {
        node: {
          id: '58aa600c-e761-5ebc-b285-eb6d59ecc4ac',
          frontmatter: { title: 'Aries', type: 'toolbox', toolboxType: 'Signs' },
          excerpt: 'Key Phrase:  I AM Basic Nature:  Pioneer and Warrior Keywords:  courageous, independent, assertive, spontaneous, head of cosmic man, self…',
          fields: { slug: '/toolbox/aries/' },
        },
      }, {
        node: {
          id: 'e9866973-380c-54ee-96c4-6a44b09dc958',
          frontmatter: { title: 'First House', type: 'toolbox', toolboxType: 'Houses' },
          excerpt: 'I AM -  The appearance, the basic mood in life, personality, vital energy, awareness of life, physical constitution, vitality, expression of…',
          fields: { slug: '/toolbox/first-house/' },
        },
      }, {
        node: {
          id: 'c96e763f-d670-5bef-a05c-3792d86a0a60',
          frontmatter: { title: 'Sun', type: 'toolbox', toolboxType: 'Planets' },
          excerpt: 'Everything in the Solar System revolves around the Sun. Growth, self-expression, outer-world activities Positive energy in any endeavor Tap…',
          fields: { slug: '/toolbox/sun/' },
        },
      }, {
        node: {
          id: 'f1fc71be-24db-5613-991a-0fdc62d5e03b',
          frontmatter: { title: 'Moon', type: 'toolbox', toolboxType: 'Planets' },
          excerpt: 'In our Solar System, the Moon reflects the light. Fundamental energy tied to emotions and intuition Celestial power tied to the Goddess Moon…',
          fields: { slug: '/toolbox/moon/' },
        },
      }, {
        node: {
          id: 'a8eabb89-ed90-5b1b-a942-80aeee38994f',
          frontmatter: { title: 'Second House', type: 'toolbox', toolboxType: 'Houses' },
          excerpt: 'I HAVE  - Assets, foundation of material, intellectual, and emotional security, wealth, possessions, finances, social background, money…',
          fields: { slug: '/toolbox/second-house/' },
        },
      }, {
        node: {
          id: '693f6b38-1f10-5555-b6f8-43cbaac2d26e',
          frontmatter: { title: 'Taurus', type: 'toolbox', toolboxType: 'Signs' },
          excerpt: 'Key Phrase:  I HAVE Basic Nature:  Builder and Accumulator Keywords:  courageous, independent, assertive, spontaneous, head of cosmic man…',
          fields: { slug: '/toolbox/taurus/' },
        },
      }, {
        node: {
          id: 'bbd334c8-9405-5f26-98d9-9617ebf82874',
          frontmatter: { title: 'Gemini', type: 'toolbox', toolboxType: 'Signs' },
          excerpt: 'Key Phrase:  I THINK Basic Nature:  Communicator and Threshold Walker Keywords:  intellectual, communicative, curious, versatile, lower neck…',
          fields: { slug: '/toolbox/gemini/' },
        },
      }, {
        node: {
          id: '80acb630-2187-5267-a599-c4359b0d68a3',
          frontmatter: { title: 'Mercury', type: 'toolbox', toolboxType: 'Planets' },
          excerpt: 'Mercury is the messenger of the Gods. Rules speech and thinking Tap energy when you use affirmations, sigils, hand gestures, and thought…',
          fields: { slug: '/toolbox/mercury/' },
        },
      },
    ],
  },
};
