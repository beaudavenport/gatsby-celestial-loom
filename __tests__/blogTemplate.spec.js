import { StaticQuery, useStaticQuery } from 'gatsby';
import React from 'react';

import renderer from 'react-test-renderer';

import BlogTemplate from '../src/templates/blogTemplate';
import blogArchiveEntries from '../__fixtures__/blogArchiveEntries';
import blogPageEntry from '../__fixtures__/blogPageEntry';
import sidebarEntries from '../__fixtures__/sidebarEntries';
import site from '../__fixtures__/site';
import siteSearchIndex from '../__fixtures__/siteSearchIndex';
import toolboxEntries from '../__fixtures__/toolboxEntries';

beforeEach(() => {
  global.window.matchMedia = jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
  }));

  StaticQuery.mockImplementation(({ render }) => render({
    ...site,
    ...siteSearchIndex,
    ...blogArchiveEntries,
    ...sidebarEntries,
  }));

  useStaticQuery.mockReturnValueOnce(toolboxEntries);
});

describe('Blog Template', () => {
  it('should render correctly with posts archive, blog page and SEO component', () => {
    const tree = renderer
      .create(<BlogTemplate data={blogPageEntry} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
