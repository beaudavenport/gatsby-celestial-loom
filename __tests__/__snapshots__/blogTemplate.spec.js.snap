// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog Template should render correctly with posts archive, blog page and SEO component 1`] = `
<div>
  <header
    className="md-paper md-paper--2 md-toolbar md-background--primary md-toolbar--text-white md-toolbar--fixed"
    style={
      Object {
        "backgroundColor": "rgb(47, 47, 47)",
      }
    }
  >
    <button
      className="md-btn md-btn--icon md-pointer--hover md-inline-block md-btn--toolbar md-toolbar--action-left"
      disabled={false}
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <div
        className="md-ink-container"
      />
      <i
        className="md-icon material-icons md-text--inherit"
      >
        menu
      </i>
    </button>
    <h2
      className="md-title md-title--toolbar"
      style={
        Object {
          "fontFamily": "Berkshire Swash",
          "fontStyle": "italic",
        }
      }
    >
      Posts
    </h2>
    <div
      className="md-cell--right md-toolbar--action-right"
    >
      <div
        className="md-btn--toolbar header-icons-container"
      >
        <div
          className="header-search-container"
        >
          <button
            className="md-btn md-pointer--hover md-text md-inline-block"
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
              }
            }
            type="button"
          >
            <div
              className="md-ink-container"
            />
            <div
              className="md-badge-container md-inline-block"
            >
              <i
                className="md-icon material-icons"
                style={
                  Object {
                    "fontSize": "2rem",
                  }
                }
              >
                shopping_cart
              </i>
            </div>
          </button>
        </div>
        <div
          className="header-search-container"
        >
          <button
            className="md-btn md-pointer--hover md-text md-inline-block"
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
              }
            }
            type="button"
          >
            <div
              className="md-ink-container"
            />
            <i
              className="md-icon material-icons"
              style={
                Object {
                  "fontSize": "2rem",
                }
              }
            >
              search
            </i>
          </button>
        </div>
      </div>
    </div>
  </header>
  <main
    className="md-navigation-drawer-content md-navigation-drawer-content--inactive md-transition--acceleration md-toolbar-relative"
    id="main-content"
    tabIndex={-1}
  >
    <div
      className="md-grid md-grid--no-spacing"
      style={
        Object {
          "maxWidth": 1100,
        }
      }
    >
      <div
        className="md-cell md-cell--8 main-contents"
      >
        <div
          className="link--container false"
        >
          <a
            href="/posts"
            style={
              Object {
                "textDecoration": "none",
              }
            }
          >
            <div
              className="touchable-link "
              style={Object {}}
            >
              <div
                className="md-ink-container"
              />
              <div
                className="flex-center"
              >
                <i
                  className="md-icon material-icons"
                  style={
                    Object {
                      "color": "#ec6602",
                      "fontSize": "2rem",
                      "marginRight": 5,
                    }
                  }
                >
                  arrow_back
                </i>
                <p
                  className="back-title"
                >
                  All Posts
                </p>
              </div>
            </div>
          </a>
        </div>
        <div
          style={
            Object {
              "padding": "20px 10px",
            }
          }
        >
          <h2
            style={
              Object {
                "fontWeight": "bold",
                "marginBottom": 5,
              }
            }
          >
            NORWAC - I AM ON MY WAY
          </h2>
          <h3
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontSize": "1.1rem",
                "marginBottom": 20,
              }
            }
          >
            May 23, 2019
          </h3>
          <div
            style={
              Object {
                "margin": "0 auto",
                "maxWidth": 400,
              }
            }
          >
            <img
              alt="post"
              src="/assets/cancer-norwac.jpg"
              style={
                Object {
                  "margin": "0 auto",
                  "objectFit": "cover",
                  "width": "100%",
                }
              }
            />
          </div>
          <div
            style={
              Object {
                "padding": 20,
              }
            }
          >
            <p
              style={
                Object {
                  "color": "#ec6602",
                  "fontWeight": "bold",
                }
              }
            >
              Mentioned in this post:
            </p>
            <div
              style={
                Object {
                  "display": "flex",
                  "flexWrap": 1,
                  "justifyContent": "flex-start",
                }
              }
            >
              <button
                className="md-chip md-chip--avatar"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                    "margin": 10,
                  }
                }
                type="button"
              >
                <div
                  className="md-inline-block md-avatar md-avatar--default md-avatar--icon-sized"
                >
                  <div
                    className="md-avatar-content"
                  >
                    <span
                      style={
                        Object {
                          "fontFamily": "AstroGadget",
                        }
                      }
                    >
                      b
                    </span>
                  </div>
                </div>
                <span
                  className="md-chip-text"
                >
                  Taurus
                </span>
              </button>
              <button
                className="md-chip md-chip--avatar"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                    "margin": 10,
                  }
                }
                type="button"
              >
                <div
                  className="md-inline-block md-avatar md-avatar--default md-avatar--icon-sized"
                >
                  <div
                    className="md-avatar-content"
                  >
                    <span
                      style={
                        Object {
                          "fontFamily": "AstroGadget",
                        }
                      }
                    >
                      1
                    </span>
                  </div>
                </div>
                <span
                  className="md-chip-text"
                >
                  First House
                </span>
              </button>
            </div>
          </div>
          <section
            className="md-card-text"
          >
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<p>I'm on my way to NORWAC - an astrological convention in Seattle, WA</p>",
                }
              }
            />
          </section>
        </div>
      </div>
      <div
        className="md-cell md-cell--4 md-cell--8-tablet sidebar-contents"
      >
        <div
          style={
            Object {
              "marginBottom": 30,
              "marginTop": 20,
            }
          }
        >
          <div
            className="sidebar-header"
          >
            <h5
              className="sidebar-header--title"
            >
              Post Archives
            </h5>
          </div>
          <div>
            <ul
              className="md-list"
            >
              <div>
                <li
                  className="md-list-item md-list-item--nested-container"
                  style={Object {}}
                >
                  <div
                    aria-expanded={false}
                    aria-pressed={false}
                    className="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-list-tile md-list-tile--two-lines md-text"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <div
                      className="md-ink-container"
                    />
                    <div
                      className="md-tile-content"
                    >
                      <div
                        className="md-tile-text--primary md-text"
                        style={
                          Object {
                            "fontWeight": "bold",
                          }
                        }
                      >
                        May 2019
                      </div>
                      <div
                        className="md-tile-text--secondary md-text--secondary"
                        style={
                          Object {
                            "fontWeight": "bold",
                          }
                        }
                      >
                        2 entries
                      </div>
                    </div>
                    <div
                      className="md-tile-addon md-tile-addon--icon"
                    >
                      <i
                        className="md-icon material-icons md-collapser"
                      >
                        keyboard_arrow_down
                      </i>
                    </div>
                  </div>
                </li>
              </div>
            </ul>
          </div>
        </div>
        <div>
          <div
            className="sidebar-header"
          >
            <h5
              className="sidebar-header--title"
            >
              Latest Updates
            </h5>
          </div>
          <div
            style={
              Object {
                "padding": 10,
              }
            }
          >
            <a
              href="/events/moon-circle/"
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <div
                className="touchable-link thumbnail-card"
                style={Object {}}
              >
                <div
                  className="md-ink-container"
                />
                <div
                  className="thumbnail-image"
                  style={
                    Object {
                      "backgroundImage": "url(/assets/gemini-moon-circle.jpg)",
                    }
                  }
                />
                <div
                  className="thumbnail-text"
                >
                  <p
                    className="thumbnail-text--subtitle"
                    style={
                      Object {
                        "color": "#ec6602",
                        "fontSize": "1rem",
                      }
                    }
                  >
                    Thursday, May 30
                  </p>
                  <p
                    className="thumbnail-text--title"
                  >
                    Moon Circle
                  </p>
                  <p
                    className="thumbnail-text--subtitle"
                  >
                    Divine Inspirations • 7 PM - 9 PM
                  </p>
                  <p
                    className="thumbnail-text--subtitle"
                  />
                </div>
              </div>
            </a>
            <a
              href="/events/stones-of-fire/"
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <div
                className="touchable-link thumbnail-card"
                style={Object {}}
              >
                <div
                  className="md-ink-container"
                />
                <div
                  className="thumbnail-image"
                  style={
                    Object {
                      "backgroundImage": "url(/assets/stones-of-fire-lotus.jpg)",
                    }
                  }
                />
                <div
                  className="thumbnail-text"
                >
                  <p
                    className="thumbnail-text--subtitle"
                    style={
                      Object {
                        "color": "#ec6602",
                        "fontSize": "1rem",
                      }
                    }
                  >
                    Saturday, Jun 08
                  </p>
                  <p
                    className="thumbnail-text--title"
                  >
                    STONES OF FIRE
                  </p>
                  <p
                    className="thumbnail-text--subtitle"
                  >
                    DeMun Healing • 1PM - 4 PM
                  </p>
                  <p
                    className="thumbnail-text--subtitle"
                  />
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "backgroundColor": "rgb(47, 47, 47)",
          "padding": 30,
        }
      }
    >
      <h6
        style={
          Object {
            "color": "white",
            "marginBottom": 0,
            "textAlign": "center",
          }
        }
      >
        Copyright 2019, The Celestial Loom. Cover photography by Photo by Anastasia Dulgier on Unsplash. Built with Gatsby and Netflify CMS.
      </h6>
    </div>
  </main>
</div>
`;
